.github/actions/secure-builder-checkout/action.yaml

```diff
@@ -0,0 +1,33 @@
+name: "secure-builder-checkout"
+description: "Checkout the builder repository"
+
+inputs:
+  repository:
```

Comment:
I feel like secure-builder-checkout could call `detect-workflow` itself since the `repository` and `ref` here are always the results of `detect-workflow`.

That way we won't even need the `repository` and `ref` inputs.

Reply:
discussed offline: until now we could have done that. But this PR requires the top re-usable workflow to checkout the builder code, so we need the repository / ref information to be available.